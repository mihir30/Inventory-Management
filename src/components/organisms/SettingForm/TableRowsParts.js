function TableRowsParts({ rowsData, deleteTableRows, handleChange, handleCheckBox }) {
  return rowsData.map((data, index) => {
    const { partName, name, required } = data;
    return (
      <tr key={index}>
        <td>
          <input
            type="text"
            key={partName}
            name="partName"
            className="part-name-input-table"
            defaultValue={partName}
            autoFocus="autoFocus"
            onChange={(e) => handleChange(index, e)}
          />
        </td>
        <td className="name-checkbox">
          <input
            type="checkbox"
            name="name"
            aria-label="label"
            defaultChecked={name}
            onChange={(e) => handleCheckBox(index, e)}
          />
        </td>
        <td className="required-checkox">
          <input
            type="checkbox"
            name="required"
            aria-label="label"
            defaultChecked={required}
            onChange={(e) => handleCheckBox(index, e)}
          />
        </td>
        <td>
          <img
            src="https://cdn3.iconfinder.com/data/icons/check-out-vol-2-glyph/48/Sed-39-256.png"
            alt="new"
            className="delete-property"
            onClick={deleteTableRows}
          />
        </td>
      </tr>
    );
  });
}
export default TableRowsParts;
