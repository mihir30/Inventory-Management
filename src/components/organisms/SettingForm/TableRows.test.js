import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ReactDom from "react-dom";
import TableRowsParts from './TableRowsParts';

describe("Table rows parts section Component", () => {
    
    const rowsData = [];
    const deleteTableRows = jest.fn();
    const handleChange = jest.fn();
    const handleCheckBox = jest.fn();

    it('renders the component properly', () => {
        const trEl = document.createElement("tr");
        ReactDom.render(<BrowserRouter><TableRowsParts 
            rowsData={rowsData} 
            deleteTableRows={deleteTableRows} 
            handleChange={handleChange} 
            handleCheckBox={handleCheckBox}>
            </TableRowsParts></BrowserRouter>, trEl); 
        
    });

    it('should have the right table rows parts component snapshot', () => {
        const tree = renderer.create().toJSON();
        expect(tree).toMatchSnapshot();
    })

});