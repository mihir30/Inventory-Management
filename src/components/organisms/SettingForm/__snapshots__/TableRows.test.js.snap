// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table rows parts section Component should have the right table rows parts component snapshot 1`] = `null`;
